networkit/cpp/components/ConnectedComponents.cpp

```diff
@@ -62,14 +62,45 @@ cc_result ConnectedComponents::get_raw_partition(const Graph & G) {
         }
     });
 
-    auto mapping_sizes = new node[n_components];
+    // add one to reserve memory for additional last offset (size)
+    auto mapping_sizes = new node[n_components + 1];
     std::fill_n(mapping_sizes, n_components, 0);
+
     for(node n = 0; n < max_id; ++n) {
         ++mapping_sizes[mapping_array[n]];
     }
+    // set size of component 0 to 0
+    mapping_sizes[0] = 0;
+
+    // sizes:   1, 2, 5, 6, 3
+    // offsets: 0, 1, 3, 8, 14
+    node sum = 0;
+    // reuse mapping_sizes as equivalence class offsets
+    auto offsets = mapping_sizes;
+    for(node component_id = 0; component_id < n_components; ++component_id) {
+        auto tmp = sum;
+        sum += mapping_sizes[component_id];
+        offsets[component_id] = tmp;
+    }
+    auto equivalence_classes = new node[max_id];
+    // reinterpret offsets as index and create equivalence classes
+    // within a class nodes are sorted
+    auto in_class_index = offsets;
```

Comment:
Why don't we copy here? For memory efficiency?

Reply:
I decided to do so for memory efficiency.